agora-sample/Sample/Governor.hs

```diff
@@ -0,0 +1,622 @@
+                        , stakeOutputDatum
+                        ]
+              , txInfoId = "1ffb9669335c908d9a4774a4bf7aa7bfafec91d015249b4138bc83fde4a3330a"
+              }
+        , scriptContextPurpose = Spending ownInputRef
+        }
+
+{- This script context should be a valid transaction for minting authority for the effect scrips.
+
+    The following components will run:
+
+    - Governor validator
+    - Authority policy
+    - Proposal validator
+
+    There should be only one proposal the transaction.
+    The validity of the proposal will be checked:
+
+    - It's in 'Agora.Proposal.Locked' state.
+    - It has a 'winner' effect group, meaning that the votes meet the requirements.
+
+    The system will ensure that for every effect scrips in said effect group,
+    a newly minted GAT is sent to the corresponding effect, and properly tagged.
+-}
+mintGATs :: ScriptContext
+mintGATs =
+  let pst = Value.singleton proposalPolicySymbol "" 1
+      gst = Value.assetClassValue govAssetClass 1
+      gat = Value.assetClassValue atAssetClass 1
+
+      ---
+
+      mockEffect :: Validator
+      mockEffect = mkValidator $ noOpValidator ""
+      mockEffectHash :: ValidatorHash
+      mockEffectHash = validatorHash mockEffect
+      mockEffectAddress :: Address
+      mockEffectAddress = scriptHashAddress mockEffectHash
+      mockEffectOutputDatum :: Datum
+      mockEffectOutputDatum = unitDatum
+      atTokenName :: TokenName
+      atTokenName = TokenName hash
+        where
+          ValidatorHash hash = mockEffectHash
+      atAssetClass :: AssetClass
+      atAssetClass = AssetClass (authorityTokenSymbol, atTokenName)
+
+      ---
+
+      governorInputDatum' :: GovernorDatum
+      governorInputDatum' =
+        GovernorDatum
+          { proposalThresholds = defaultProposalThresholds
+          , nextProposalId = ProposalId 5
+          }
+      governorInputDatum :: Datum
+      governorInputDatum = Datum $ toBuiltinData governorInputDatum'
+      governorInput :: TxOut
+      governorInput =
+        TxOut
+          { txOutAddress = govValidatorAddress
+          , txOutValue = gst
+          , txOutDatumHash = Just $ toDatumHash governorInputDatum
+          }
+
+      ---
+
+      effects =
+        AssocMap.fromList
+          [ (ResultTag 0, AssocMap.empty)
+          , (ResultTag 1, AssocMap.singleton mockEffectHash $ toDatumHash mockEffectOutputDatum)
+          ]
+      proposalVotes :: ProposalVotes
+      proposalVotes =
+        ProposalVotes $
+          AssocMap.fromList
+            [ (ResultTag 0, 100)
+            , (ResultTag 1, 2000) -- The winner
+            ]
+      proposalInputDatum' :: ProposalDatum
+      proposalInputDatum' =
+        ProposalDatum
+          { P.proposalId = ProposalId 0
+          , effects = effects
+          , status = Locked
+          , -- TODO: Any need to check minimun amount of cosigners here?
+            cosigners = [signer, signer2]
```

Comment:
I don't think it's important, since it would be impossible to get to `status == Locked` otherwise . P.S. We might have to drop cosignature implementation until after preaudit. So let's not worry about it too much for now.